frontend/src/metabase/visualizations/components/legend/LegendCaption.tsx

```diff
@@ -76,19 +80,22 @@ export const LegendCaption = ({
   return (
     <LegendCaptionRoot className={className} data-testid="legend-caption">
       {icon && <LegendLabelIcon {...icon} />}
-      <LegendLabel
-        className={cx(
-          DashboardS.fullscreenNormalText,
-          DashboardS.fullscreenNightText,
-          EmbedFrameS.fullscreenNightText,
-        )}
-        href={href}
-        onClick={onSelectTitle}
-        onFocus={handleFocus}
-        onMouseEnter={handleMouseEnter}
-      >
-        <Ellipsified data-testid="legend-caption-title">{title}</Ellipsified>
-      </LegendLabel>
+      <Group gap="0.25rem" align="center">
+        <LegendLabel
+          className={cx(
+            DashboardS.fullscreenNormalText,
+            DashboardS.fullscreenNightText,
+            EmbedFrameS.fullscreenNightText,
+          )}
+          href={href}
+          onClick={onSelectTitle}
+          onFocus={handleFocus}
+          onMouseEnter={handleMouseEnter}
+        >
+          <Ellipsified data-testid="legend-caption-title">{title}</Ellipsified>
+        </LegendLabel>
+        {card && <PLUGIN_MODERATION.EntityModerationIcon card={card} filled />}
```

Comment:
Could we make this legend and viz components have a slot like `legendRightSection` here so the viz layer does not know about moderation and enterprise plugins? And this icon would be passed from the pinned items list in collections